### PR TITLE
update svelte skeleton

### PR DESCRIPTION
**Pull request by tosinek**
The Svelte integration seems to have issues when both `zodern:melte` and `rdb:svelte-meteor-data` are used together. It is quite random, some stuff will just stop being reactive, while some other still works. I think it was even breaking HMR in a Blaze+Svelte project for me. When using just one package, all the problems suddenly disappear.

I suggest updating the skeleton to only include zodern:melte, which is also aligned with the guide at https://svelte-tutorial.meteor.com/

Related issues:
https://github.com/zodern/melte/issues/24
https://github.com/zodern/melte/issues/25

I also updated the skeleton to utilize the collection. Not really necessary, but if it is not used, then it might be removed completely instead?

### CODE REVIEW

**Comment by tosinek** [comment]
I can see this table would need updating too
https://docs.meteor.com/commandline.html#meteorcreate

**Comment by Grubba27** [comment]
Loved this change! Update the docs(that table). 
In the next meteor patch release I will make sure that this comes up too

**Review by Grubba27 — APPROVED** [review]
LGTM!

File: tools/static-assets/skel-svelte/imports/ui/App.svelte [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
   let counter = 0;
   const addToCounter = () => {
     counter += 1;
   }
+  
+  const subIsReady = true; // remove this line if you want to use the code below
+  // no need to publish/subscribe as there is autopublish package installed
+  // let subIsReady = false;
+  // $m: {
+  //   const handle = Meteor.subscribe('links.all'); // todo: setup the server-side publication
```
Comment (zodern) [inline tools/static-assets/skel-svelte/imports/ui/App.svelte:13]: There doesn't seem to be any server code for this publication.

File: tools/static-assets/skel-svelte/imports/ui/App.svelte [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  //   subIsReady = handle.ready();
+  // }
```
[the remark this answers is not in the document]
Comment (zodern) [reply tools/static-assets/skel-svelte/imports/ui/App.svelte:15]: It might also be good to add the server code for the publication.

**Comment by tosinek** [comment]
> I've been wondering if we should configure the svelte skeleton so it supports typescript. It doesn't need to have any typescript code, but at least have the tsconfig and dependencies installed so you don't have to do anything to start using typescript.

Great idea. I am thinking about a whole guide for Svelte, like the one for React https://guide.meteor.com/react.html to document TS, SSR, i18n, reactivity, blaze integration or routing

**Comment by Grubba27** [comment]
Is there something missing? @tosinek ? can I merge this to 2.9.1?

**Comment by Grubba27** [comment]
IMO @tosinek, I would convert it to TypeScript, the same way we have with React-TS.
By default we should enforce typesafety

**Comment by tosinek** [comment]
Ok, few more minutes :)

**Comment by Grubba27** [comment]
Ci wont work but I will take a look in the 2.9.1 branch. Just let me know when you are done :)

**Comment by tosinek** [comment]
I think I am done. The server/main.js could be still converted, but may leave it to another update

**Comment by Grubba27** [comment]
You can copy from the other skeleton

**Comment by tosinek** [comment]
I think it is time to merge it as it is. Importing .svelte files in .ts files is making TS to complain, so I will have to figure it out, but not tonight.

**Comment by Grubba27** [comment]
Got it! Thx for your work @tosinek! hope to see you again :)

**Comment by tosinek** [comment]
I wanna create a Svelte guide. Would it make sense to have routing in this skeleton? It is something I was missing a lot in the beginning. I can see that the new Vue skeleton includes routing. Maybe there could even be some kind of guideline on what to include in every skeleton? (eg. that the we are trying to enforce type safety)

**Comment by Grubba27** [comment]
We want to maintain a standard for every skeleton, so I think this is a nice one. Most apps will need a router! I make issues so the community can help with other frameworks.

Also, while talking with the team, we decided that for now, it could be nice to have the tsconfig.json already done so that to make your app a TypeScript, one would just need to add the ts files. I will make the change, for now, to revert the ``links.ts`` to ``links.js``.

**Comment by tosinek** [comment]
I think .ts works fine for this skeleton, except for the main client entrypoint that is still .js. It does not like the App.svelte import there. Links.ts work fine on my machine. 

And I created a minimalistic tsconfig.json with comments that seems to be working (included in the PR). Because it is difficult for me to use a long config with many options I don't even understand (eg the one from the Meteor guide) without knowing what is important there and what is just a preference.

**Pull request merged**